db/migrate/20230214025901_update_reporting_events_with_incorrect_first_responses.rb

```diff
@@ -0,0 +1,45 @@
+class UpdateReportingEventsWithIncorrectFirstResponses < ActiveRecord::Migration[6.1]
+  include ReportingEventHelper
+
+  # rubocop:disable Metrics/AbcSize
+  def change
+    ReportingEvent.where(name: 'first_response', user_id: nil).find_in_batches do |bot_responses|
+      bot_responses.each do |event|
+        conversation = event.conversation
+        next if conversation.nil?
+
+        last_bot_reply = conversation.messages.where(sender_type: 'AgentBot').order(created_at: :asc).last
```

Comment:
We have over a million records in our cloud. Running this migration sequentially wouldn't be effective. Instead, lets move the logic to data migration sidekiq jobs.  lets go with this strategy 

- iterate over accounts and enqueue a recompute job for each account
- the logic in this migration happens in the recompute job for scoped for that account.  

ref: https://github.com/chatwoot/chatwoot/pull/4576/files

Reply:
This commit resolves it https://github.com/chatwoot/chatwoot/pull/6409/commits/1992fc2c187bf3e8ef2024bb69ae33b5eb6b0253 PTAL